x/concentrated-liquidity/store.go

```diff
@@ -101,3 +102,71 @@ func ParseFullPositionFromBytes(key, value []byte) (types.FullPositionByOwnerRes
 		Liquidity:   positionValue.Liquidity,
 	}, nil
 }
+
+// ParseFullIncentiveRecordFromBz parses a IncentiveRecord from a byte array.
+// Returns a struct containing the denom and min uptime associated with the incentive record.
+// Returns an error if the byte array is empty.
+// Returns an error if fails to parse.
+func ParseIncentiveRecordFromBz(bz []byte) (incentiveRecord types.IncentiveRecord, err error) {
+	if len(bz) == 0 {
+		return types.IncentiveRecord{}, errors.New("incentive record not found")
+	}
+	err = proto.Unmarshal(bz, &incentiveRecord)
+	return incentiveRecord, err
```

Comment:
shouldn't we catch the error here to ensure if an error occurs that the incentive record that gets returned is empty? (I assume this is done by Unmarshal itself but seems just nice to guarantee on our end)

Reply:
Unmarshal does handle this, but I agree that for readability it's better to catch the error. Added!